src/Nox.ClientApi.Tests/Tests/Controllers/StoresControllerTests.cs

```diff
@@ -143,6 +144,142 @@ public async Task Get_StoreOwnerOdataQuery_ReturnOwner()
 
         #endregion GET Expand Relation /api/{EntityPluralName}/{EntityKey} => api/stores/1?$expand=Ownership
 
+        #region POST Entity with RelationshipId /api/{EntityPluralName} => api/stores
+        [Fact]
+        public async Task Post_WithRelationshipId_CreatesRefToRelatedEntity()
+        {
+            // Arrange
+            var licenseName = _fixture.Create<string>();
+            var licenseCreateDto = new StoreLicenseCreateDto
+            {
+                Issuer = licenseName
+            };
+            var licensePostResponse = await PostAsync<StoreLicenseCreateDto, StoreLicenseDto>(StoreLicenseUrl, licenseCreateDto);
+
+            var storeCreateDto = new StoreCreateDto
+            {
+                Name = _fixture.Create<string>(),
+                Address = new StreetAddressDto(
+                    StreetNumber: null!,
+                    AddressLine1: "3000 Hillswood Business Park",
+                    AddressLine2: null!,
+                    Route: null!,
+                    Locality: null!,
+                    Neighborhood: null!,
+                    AdministrativeArea1: null!,
+                    AdministrativeArea2: null!,
+                    PostalCode: "KT16 0RS",
+                    CountryId: CountryCode.GB),
+                Location = new LatLongDto(51.3728033, -0.5389749),
+                LicenseId = licensePostResponse!.Id
+            };
+
+            // Act
+            var result = await PostAsync<StoreCreateDto, StoreDto>(EntityUrl, storeCreateDto);
+
+            const string oDataRequest = $"$expand={nameof(StoreDto.License)}";
+            var response = await GetODataSimpleResponseAsync<StoreDto>($"{EntityUrl}/{result!.Id}?{oDataRequest}");
+
+            //Assert
+            result.Should().NotBeNull();
+            result.Should()
+                .BeOfType<StoreDto>()
+                .Which.Id.Should().NotBeEmpty();
+            response.Should().NotBeNull();
+            response!.License.Should().NotBeNull();
+            response!.License!.Id.Should().Be(licensePostResponse!.Id);
+            response!.License!.Issuer.Should().Be(licenseName);
+        }
+        #endregion
+
+        #region POST Entity with Invalid RelationshipId /api/{EntityPluralName} => api/stores
+        [Fact]
+        public async Task Post_WithInvalidRelationshipId_CreatesEntityWithoutRelationship()
```

Comment:
@rochar  if LicenseId has invalid Id then Store will be created anyway, License will be null
Is it what we want? Or do we want to throw an exception?

Reply:
we want to throw a custom exception